tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -115,12 +116,12 @@ public function testAssignInIf()
 				$this->assertArrayHasKey('anotherNullableIntegerFromTryCatch', $variables);
 				$this->assertSame('int|null', $variables['anotherNullableIntegerFromTryCatch']->describe());
 
-				$this->assertSame('mixed[]', $variables['nullableIntegers']->describe());
-				$this->assertSame('mixed[]', $variables['mixeds']->describe());
+				$this->assertSame('(int|null)[]', $variables['nullableIntegers']->describe());
+				$this->assertSame('(int|string)[]', $variables['union']->describe());
 
-				/** @var \PHPStan\Type\ArrayType $mixeds */
-				$mixeds = $variables['mixeds'];
-				$this->assertInstanceOf(MixedType::class, $mixeds->getItemType());
+				/** @var \PHPStan\Type\ArrayType $union */
+				$union = $variables['union'];
+				$this->assertInstanceOf(UnionType::class, $union->getItemType());
```

Comment:
That is already done on line 120. I replace the union assert with

~~~php
$this->assertSame('int|string', $variables['union']->getIterableValueType()->describe());
~~~